tests/integration/concrete/batch/batch.t.sol

```diff
@@ -0,0 +1,165 @@
+            lockup.createWithDurationsLD, (defaults.createWithDurations(), defaults.segmentsWithDurations())
+        );
+        calls[1] = abi.encodeCall(
+            lockup.createWithDurationsLL,
+            (defaults.createWithDurations(), defaults.unlockAmounts(), defaults.durations())
+        );
+        calls[2] = abi.encodeCall(
+            lockup.createWithDurationsLT, (defaults.createWithDurations(), defaults.tranchesWithDurations())
+        );
+        calls[3] = abi.encodeCall(lockup.createWithTimestampsLD, (defaults.createWithTimestamps(), defaults.segments()));
+        calls[4] = abi.encodeCall(
+            lockup.createWithTimestampsLL,
+            (defaults.createWithTimestamps(), defaults.unlockAmounts(), defaults.CLIFF_TIME())
+        );
+        calls[5] = abi.encodeCall(lockup.createWithTimestampsLT, (defaults.createWithTimestamps(), defaults.tranches()));
+
+        // It should return the stream IDs created.
+        bytes[] memory results = lockup.batch{ value: 1 wei }(calls);
+        assertEq(results.length, 6, "batch results length");
+        assertEq(abi.decode(results[0], (uint256)), expectedNextStreamId, "batch results[0]");
+        assertEq(abi.decode(results[1], (uint256)), expectedNextStreamId + 1, "batch results[1]");
+        assertEq(abi.decode(results[2], (uint256)), expectedNextStreamId + 2, "batch results[2]");
+        assertEq(abi.decode(results[3], (uint256)), expectedNextStreamId + 3, "batch results[3]");
+        assertEq(abi.decode(results[4], (uint256)), expectedNextStreamId + 4, "batch results[4]");
+        assertEq(abi.decode(results[5], (uint256)), expectedNextStreamId + 5, "batch results[5]");
+        assertEq(address(lockup).balance, initialEthBalance + 1 wei, "batch contract balance");
+    }
+
+    /// @dev The batch call includes:
+    /// - ETH value
+    /// - All recipient related functions with both returns and non-returns
+    function test_BatchPayable_RecipientFunctions() external {
+        uint256 initialEthBalance = address(lockup).balance;
+        vm.warp(defaults.WARP_26_PERCENT());
+
+        bytes[] memory calls = new bytes[](4);
+        calls[0] = abi.encodeCall(lockup.cancel, (defaultStreamId));
+
+        uint256[] memory streamIds = new uint256[](2);
+        streamIds[0] = recipientGoodStreamId;
+        streamIds[1] = recipientInvalidSelectorStreamId;
+        calls[1] = abi.encodeCall(lockup.cancelMultiple, (streamIds));
+
+        calls[2] = abi.encodeCall(lockup.renounce, (recipientReentrantStreamId));
+
+        streamIds = new uint256[](1);
+        streamIds[0] = recipientRevertStreamId;
+        calls[3] = abi.encodeCall(lockup.renounceMultiple, (streamIds));
+
+        bytes[] memory results = lockup.batch{ value: 1 wei }(calls);
+
+        assertEq(results.length, 4);
+        assertEq(results[0], hex"");
+        assertEq(results[1], hex"");
+        assertEq(results[2], hex"");
+        assertEq(results[3], hex"");
```

Comment:
we don't need to specifically add `hex`

same above & below

Reply:
How would you compare that the result is empty then?

Reply:
`assertEq(results[0], "");`

Reply:
Thanks for the suggestion, I tested and it works. I remember trying it but it was not working which led me to add `hex`. May be I did not try this but `0x`.